homeassistant/util/logging.py

```diff
@@ -18,6 +20,60 @@
     get_hassjob_callable_job_type,
 )
 
+_LOGGER = logging.getLogger(__name__)
+
+
+class HomeAssistantQueueListener(logging.handlers.QueueListener):
+    """Custom QueueListener to watch for noisy loggers."""
+
+    LOG_COUNTS_RESET_INTERVAL = 300
+    MAX_LOGS_COUNT = 200
+
+    _last_reset: float
+    _log_counts: dict[str, int]
+    _warned_modules: set[str]
+
+    def __init__(
+        self, queue: SimpleQueue[logging.Handler], *handlers: logging.Handler
+    ) -> None:
+        """Initialize the handler."""
+        super().__init__(queue, *handlers)
+        self._warned_modules = set()
+        self._reset_counters(time.time())
+
+    @override
+    def handle(self, record: logging.LogRecord) -> None:
+        """Handle the record."""
+        super().handle(record)
+
+        if record.levelno < logging.INFO:
+            return
+
+        if (record.created - self._last_reset) > self.LOG_COUNTS_RESET_INTERVAL:
+            self._reset_counters(record.created)
+
+        module_name = record.name
+        if module_name == __name__ or module_name in self._warned_modules:
+            return
+
+        self._log_counts[module_name] += 1
+        module_count = self._log_counts[module_name]
+        if module_count < self.MAX_LOGS_COUNT:
+            return
+
+        _LOGGER.warning(
+            "Module %s is logging too frequently. %d messages since last count",
+            module_name,
+            module_count,
+        )
+        self._log_counts[module_name] = 0
```

Comment:
Why do we reset the count for the warned module? It will never be counted again.

Reply:
It was a leftover from the initial version, but not needed now. Removed it.